src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs

```diff
@@ -0,0 +1,105 @@
+
+        /// <summary>
+        /// Note: This name is used in xaml file.
+        /// </summary>
+        private const string MultipleMembersContextMenuStyle = nameof(MultipleMembersContextMenuStyle);
+
+        public static InheritanceMargin CreateForSingleMember(
+            IThreadingContext threadingContext,
+            IStreamingFindUsagesPresenter streamingFindUsagesPresenter,
+            IWaitIndicator waitIndicator,
+            Solution solution,
+            SingleMemberMarginViewModel viewModel)
+        {
+            var margin = new InheritanceMargin(threadingContext, streamingFindUsagesPresenter, waitIndicator, solution);
+            // This is created in the xaml file.
+            margin.DataContext = viewModel;
+            var contextMenu = margin.ContextMenu!;
+            contextMenu.DataContext = viewModel;
+            contextMenu.Style = (Style)margin.FindResource(SingleMemberContextMenuStyle);
+            return margin;
+        }
+
+        public static InheritanceMargin CreateForMultipleMembers(
+            IThreadingContext threadingContext,
+            IStreamingFindUsagesPresenter streamingFindUsagesPresenter,
+            IWaitIndicator waitIndicator,
+            Solution solution,
+            MultipleMembersMarginViewModel viewModel)
+        {
+            var margin = new InheritanceMargin(threadingContext, streamingFindUsagesPresenter, waitIndicator, solution);
+            // This is created in the xaml file.
+            margin.DataContext = viewModel;
+            var contextMenu = margin.ContextMenu!;
+            contextMenu.DataContext = viewModel;
+            contextMenu.Style = (Style)margin.FindResource(MultipleMembersContextMenuStyle);
+            return margin;
+        }
+
+        private InheritanceMargin(
+            IThreadingContext threadingContext,
+            IStreamingFindUsagesPresenter streamingFindUsagesPresenter,
+            IWaitIndicator waitIndicator,
+            Solution solution)
+        {
+            _threadingContext = threadingContext;
+            _streamingFindUsagesPresenter = streamingFindUsagesPresenter;
+            _solution = solution;
+            _waitIndicator = waitIndicator;
+            InitializeComponent();
+        }
+
+        private void Margin_OnClick(object sender, RoutedEventArgs e)
+        {
+            if (this.ContextMenu != null)
+            {
+                this.ContextMenu.IsOpen = true;
+                e.Handled = true;
+            }
+        }
+
+        private void TargetMenuItem_OnClick(object sender, RoutedEventArgs e)
+        {
+            if (e.OriginalSource is MenuItem menuItem && menuItem.DataContext is TargetDisplayViewModel viewModel)
+            {
+                _waitIndicator.Wait(
+                    title: EditorFeaturesResources.Navigating,
+                    message: EditorFeaturesResources.Navigating_to_definition,
```

Comment:
ideally we could say 'navigating to {0}' so it's clear where we're going.

Reply:
changed